src/qibolab/_core/instruments/qblox/sequence/sequence.py

```diff
@@ -165,20 +181,28 @@ def compile(
     sweepers: list[ParallelSweepers],
     options: ExecutionParameters,
     sampling_rate: float,
+    offsets: dict[ChannelId, float],
     los: dict[ChannelId, OscillatorConfig],
     time_of_flights: dict[ChannelId, float],
 ) -> dict[ChannelId, Q1Sequence]:
     duration = sequence.duration
+    sweeper_channels = {
+        channel: []
+        for ps in sweepers
+        for sweeper in ps
+        for channel in (sweeper.channels if sweeper.channels is not None else [])
+    }
     return {
         ch: Q1Sequence.from_pulses(
             seq,
-            sweepers,
+            sweepers[::-1],
```

Comment:
I understand the problem, since the last sweeper is the innermost, but this means that it is coming first in the assembly.

However, I feel this is not the best place to reverse it, since then it would be difficult to track until when it is sorted in one way (innermost last) and when it is instead intended reversed.

I will move this as close as possible to the assembly generation (most likely in `loops_machinery` - but it has to be negotiated also with the registers initialization).